proposals/readonly-setter-calls-on-non-variables.md

```diff
@@ -0,0 +1,127 @@
+# Readonly setter calls on non-variables
+
+Champion issue: <https://github.com/dotnet/csharplang/issues/9174>
+
+## Summary
+
+Permits a readonly setter to be called on _all_ non-variable expressions:
+
+```cs
+var c = new C();
+// Remove the current CS1612 error, because ArraySegment<T>.this is readonly:
+c.ArraySegmentProp[10] = new object();
+
+// The same restriction has *already* been lifted for invocation expressions:
+c.ArraySegmentMethod()[10] = new object();
+
+class C
+{
+    public ArraySegment<object> ArraySegmentProp { get; set; }
+    public ArraySegment<object> ArraySegmentMethod() => ArraySegmentProp;
+}
```

Comment:
Another example to consider here is a `readonly struct` that has a `ref` returning indexer. That shows another way that we support getting around this limitation. The only reason that a `readonly struct` can't universally just take that approach is that some data storage backing, like if you backed your `readonly struct` with `List<T>` vs. `T[]` is the former doesn't have a great way at getting `ref` to the values.